### PR TITLE
refactor(admin): event buttons for myself turned off in open contributions

### LINKED ISSUE #2710: 🐛 [Bug] delete own post in admin interface remove icon
## 🐛 Bugreport
If the own post cannot be deleted in the admin interface, the icon should not be visible either.

>>> Wenn der eigene Beitrag im Admin-Interface nicht gelöscht werden kann, sollte auch das Icon nicht sichtbar sein.

Bildschirmfoto_2023-02-14_um_08 35 51

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
event buttons for myself turned off in open contributions

### Issues
- fixes #2710 

image

### CODE REVIEW

File: admin/src/components/Tables/OpenCreationsTable.vue [hunk 1]
```diff
@@ -13,14 +13,16 @@
...
       <template #cell(bookmark)="row">
-        <b-button
-          variant="danger"
-          size="md"
-          @click="$emit('show-overlay', row.item, 'delete')"
-          class="mr-2"
-        >
-          <b-icon icon="trash" variant="light"></b-icon>
-        </b-button>
+        <div v-if="!myself(`${row.item.firstName} ${row.item.lastName}`)">
```
Comment (Mogge) [inline admin/src/components/Tables/OpenCreationsTable.vue:16]:
```suggestion
        <div v-if="!myself(row.item)">
```

File: admin/src/components/Tables/OpenCreationsTable.vue [hunk 2]
```diff
@@ -49,17 +51,19 @@
         </div>
       </template>
       <template #cell(reActive)>
-        <b-button variant="warning" size="md" class="mr-2">
-          <b-icon icon="arrow-up" variant="light"></b-icon>
-        </b-button>
+        <div v-if="!myself(`${row.item.firstName} ${row.item.lastName}`)">
```
Comment (Mogge) [inline admin/src/components/Tables/OpenCreationsTable.vue:54]:
```suggestion
        <div v-if="!myself(row.item)">
```

File: admin/src/components/Tables/OpenCreationsTable.vue [hunk 3]
```diff
@@ -49,17 +51,19 @@
...
+        </div>
       </template>
       <template #cell(chatCreation)="row">
         <b-button v-if="row.item.messagesCount > 0" @click="rowToggleDetails(row, 0)">
           <b-icon icon="chat-dots"></b-icon>
         </b-button>
       </template>
       <template #cell(deny)="row">
-        <div v-if="$store.state.moderator.id !== row.item.userId">
+        <div v-if="!myself(`${row.item.firstName} ${row.item.lastName}`)">
```
Comment (Mogge) [inline admin/src/components/Tables/OpenCreationsTable.vue:66]:
```suggestion
        <div v-if="!myself(row.item)">
```

File: admin/src/components/Tables/OpenCreationsTable.vue [hunk 4]
```diff
@@ -71,7 +75,7 @@
         </div>
       </template>
       <template #cell(confirm)="row">
-        <div v-if="$store.state.moderator.id !== row.item.userId">
+        <div v-if="!myself(`${row.item.firstName} ${row.item.lastName}`)">
```
Comment (Mogge) [inline admin/src/components/Tables/OpenCreationsTable.vue:78]:
```suggestion
        <div v-if="!myself(row.item)">
```

File: admin/src/components/Tables/OpenCreationsTable.vue [hunk 5]
```diff
@@ -158,6 +162,11 @@ export default {
     }
   },
   methods: {
+    myself(name) {
```
Comment (Mogge) [inline admin/src/components/Tables/OpenCreationsTable.vue:165]:
```suggestion
    myself(item) {
```

File: admin/src/components/Tables/OpenCreationsTable.vue [hunk 6]
```diff
@@ -158,6 +162,11 @@ export default {
...
+      return (
+        name === `${this.$store.state.moderator.firstName} ${this.$store.state.moderator.lastName}`
```
Comment (Mogge) [inline admin/src/components/Tables/OpenCreationsTable.vue:167]:
```suggestion
        `${item.firstName} ${item.lastName}` === `${this.$store.state.moderator.firstName} ${this.$store.state.moderator.lastName}`
```

**Review by Mogge — DISMISSED** [review]
We have to refactor this when we have the id of the user. This is not working when it is a creation of the moderator itself

**Review by ulfgebhardt — COMMENTED** [review]
Please make an issue for this refactor so we remember it!

**Review by Elweyn — DISMISSED** [review]

**Review by ulfgebhardt — APPROVED** [review]

**Review by ulfgebhardt — APPROVED** [review]

**Review by Mogge — APPROVED** [review]

**Pull request merged**